### PR TITLE
Update Metadata and Installation Documentation

### PR DESCRIPTION
**Pull request by ron-unstructured**
* Updated Metadata page: add common and additional metadata fields by document types and connectors
* Updated specific installation extra by document types and connectors
* Added embedding brick page in Sphinx TOC
* Fixed Sphinx warnings in new pages

### CODE REVIEW

File: docs/source/metadata.rst [hunk 1]
```diff
@@ -8,29 +12,67 @@ or an e-mail with a given subject line.
...
+| ``filetype``                | File Type                                                |                                                                                                                                                                                                                                                                                             |
++-----------------------------+----------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| ``type``                    | Element Type                                             | Categorizes elements into types such as Title, NarrativeText. Not a metadata field                                                                                                                                                                                                          |
```
Comment (cragwolfe) [inline docs/source/metadata.rst:35]: need to be careful here as the `type` technically lives outside of `metadata` even though downstream, for all intents and purposes, it kind of is.

https://github.com/Unstructured-IO/unstructured/blob/19d8bff/test_unstructured_ingest/expected-structured-output/s3/small-pdf-set/2023-Jan-economic-outlook.pdf.json#L3

Maybe change the first column to `N/A` and update Details to mention the caveat?

File: docs/source/metadata.rst [hunk 2]
```diff
@@ -8,29 +12,67 @@ or an e-mail with a given subject line.
...
+|                             | other elements of the same category                      | Category depth is set using native document hierarchies (e.g., h1, h2, h3 or the indentation level of a bulleted list in a word document).                                                                                                                                                  |
++-----------------------------+----------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| ``text_as_html``            | HTML representation of extracted tables                  | Elements with type Table                                                                                                                                                                                                                                                                    |
```
Comment (cragwolfe) [inline docs/source/metadata.rst:45]:
```suggestion
| ``text_as_html``            | HTML representation of extracted tables                  |  Only applicable to ``Table`` Elements                                                                                                                                                                                                                                                                    |
```

File: docs/source/metadata.rst [hunk 3]
```diff
@@ -49,18 +91,56 @@ returned. If the ``in_place`` flag is ``False``, only the altered coordinates ar
...
+| ``header_footer_type``  | Word Doc            | Pages a header or footer applies to: "primary",        |
+|                         |                     | "even_only", and "first_page"                          |
++-------------------------+---------------------+--------------------------------------------------------+
+| ``url``                 | HTML                | Webpage URL                                            |
++-------------------------+---------------------+--------------------------------------------------------+
+| ``link_urls``           | HTML                | The url associated with a link in a document.          |
++-------------------------+---------------------+--------------------------------------------------------+
+| ``link_texts``          | HTML                | The text associated with a link in a document.         |
++-------------------------+---------------------+--------------------------------------------------------+
+| ``links``               | HTML                | List of {”text”: “<the text>, “url”: <the url>} items. |
```
Comment (cragwolfe) [inline docs/source/metadata.rst:136]:
```suggestion
| ``links``               | HTML                | List of {”text”: “<the text>, “url”: <the url>} items. This element will be removed in the near future in favor of the above two rows |
```

File: docs/source/metadata.rst [hunk 4]
```diff
@@ -8,29 +12,67 @@ or an e-mail with a given subject line.
...
+|                             | for max_characters in add_chunking_strategy              |                                                                                                                                                                                                                                                                                             |
++-----------------------------+----------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| ``is_continuation``         | True if element is a continuation of table being chunked | Used for chunking                                                                                                                                                                                                                                                                           |
```
Comment (cragwolfe) [inline docs/source/metadata.rst:56]:
```suggestion
| ``is_continuation``         | True if element is a continuation of a previous element | Only relevant for chunking, if an element was divided into two due to ``max_characters``                                                                                                                                                                                                                                                                           |
```

File: docs/source/metadata.rst [hunk 5]
```diff
@@ -8,29 +12,67 @@ or an e-mail with a given subject line.
...
+| ``type``                    | Element Type                                             | Categorizes elements into types such as Title, NarrativeText. Not a metadata field                                                                                                                                                                                                          |
++-----------------------------+----------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| ``coordinates``             | XY Bounding Box Coordinates                              |                                                                                                                                                                                                                                                                                             |
```
Comment (cragwolfe) [inline docs/source/metadata.rst:37]: Include in-doc link to further down the page where coordinates are explained in more detail?

File: docs/source/metadata.rst [hunk 6]
```diff
@@ -49,18 +91,56 @@ returned. If the ``in_place`` flag is ``False``, only the altered coordinates ar
...
 	print(element.metadata.coordinates.system.height)
 
+###########################################
+Additional Metadata Fields by Document Type
+###########################################
+
++-------------------------+---------------------+--------------------------------------------------------+
+| ``Field Name``          | Applicable Doc Types| Short Description                                      |
++=========================+=====================+========================================================+
+| ``page_number``         | PDF, HTML, PPT      | Page Number                                            |
```
Comment (cragwolfe) [inline docs/source/metadata.rst:115]:
```suggestion
| ``page_number``         | PDF, PPT, DOCX      | Page Number                                            |
```

File: docs/source/metadata.rst [hunk 7]
```diff
@@ -49,18 +91,56 @@ returned. If the ``in_place`` flag is ``False``, only the altered coordinates ar
...
+| ``sent_to``             | EML                 | Email Recipient                                        |
++-------------------------+---------------------+--------------------------------------------------------+
+| ``subject``             | EML                 | Email Subject                                          |
++-------------------------+---------------------+--------------------------------------------------------+
+| ``attached_to_filename``| MSG                 | filename that attachment file is attached to           |
++-------------------------+---------------------+--------------------------------------------------------+
+| ``header_footer_type``  | Word Doc            | Pages a header or footer applies to: "primary",        |
+|                         |                     | "even_only", and "first_page"                          |
++-------------------------+---------------------+--------------------------------------------------------+
+| ``url``                 | HTML                | Webpage URL                                            |
```
Comment (cragwolfe) [inline docs/source/metadata.rst:130]: i think url's only come from docs that came through connectors, in which case this row should be removed

File: docs/source/metadata.rst [hunk 8]
```diff
@@ -89,13 +169,87 @@ Webpages
...
 
+##############################
+Data Connector Metadata Fields
+##############################
+
+Common Data Connector Metadata Fields
+-------------------------------------
+
+- Source Metadata
+    - Source metadata includes (field on the `BaseIngestDoc` class:
```
Comment (cragwolfe) [inline docs/source/metadata.rst:180]: this is confusing, i don't think we need to expose this level of detail here

File: docs/source/metadata.rst [hunk 9]
```diff
@@ -8,29 +12,67 @@ or an e-mail with a given subject line.
...
+| ``text_as_html``            | HTML representation of extracted tables                  | Elements with type Table                                                                                                                                                                                                                                                                    |
++-----------------------------+----------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| ``languages``               | Document Languages                                       | At document level or element level                                                                                                                                                                                                                                                          |
```
Comment (qued) [inline docs/source/metadata.rst:47]:
```suggestion
| ``languages``               | Document Languages                        | At document level or element level.                                                                                                                                                    |
```

File: docs/source/metadata.rst (hunk 1 above)
Comment (shreyanid) [reply docs/source/metadata.rst:35]: +1 on `type` not actually being in metadata. I removed it from the original table since I noticed the very top of the notion doc was "Types (Categories)" - I think it's best to mention its availability separately (like [here](https://unstructured-io.github.io/unstructured/introduction.html#document-elements) in the docs and remove it from this table.

File: docs/source/metadata.rst [hunk 10]
```diff
@@ -8,29 +12,67 @@ or an e-mail with a given subject line.
...
+| ``text_as_html``            | HTML representation of extracted tables                  |  Only applicable to ``Table`` Elements                                                                                                                                                                                                                                                                    |
++-----------------------------+----------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| ``languages``               | Document Languages                                       | At document level or element level                                                                                                                                                                                                                                                          |
```
Comment (shreyanid) [inline docs/source/metadata.rst:47]:
```suggestion
| ``languages``               | Document Languages                                       | At document level or element level. List is ordered by probability of being the primary language of the text.                                                                                                                                                                                                                                                          |
```

File: docs/source/metadata.rst (hunk 4 above)
Comment (qued) [inline docs/source/metadata.rst]:
```suggestion
| ``is_continuation``         | True if element is a continuation of table being chunked | Used for chunking. |
```

File: docs/source/metadata.rst [hunk 11]
```diff
@@ -8,29 +12,67 @@ or an e-mail with a given subject line.
...
+| ``is_continuation``         | True if element is a continuation of table being chunked | Used for chunking                                                                                                                                                                                                                                                                           |
++-----------------------------+----------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| ``detection_class_prob``    | Detection Model Class Probabilities                      | From unstructured-inference, hi-res strategy                                                                                                                                                                                                                                                |
```
Comment (qued) [inline docs/source/metadata.rst:58]:
```suggestion
| ``detection_class_prob``    | Detection model class probabilities       | From unstructured-inference, hi-res strategy.                          |
```

File: docs/source/metadata.rst [hunk 12]
```diff
@@ -8,29 +12,67 @@ or an e-mail with a given subject line.
...
 If it exists, an element's location data is available with ``element.metadata.coordinates``.
 
 The ``coordinates`` property of an ``ElementMetadata`` stores:
 
-* points: These specify the corners of the bounding box starting from the top left corner and
+* **points**: These specify the corners of the bounding box starting from the top left corner and
   proceeding counter-clockwise. The points represent pixels, the origin is in the top left and
   the ``y`` coordinate increases in the downward direction.
-* system: The points have an associated coordinate system. A typical example of a coordinate system is
+* **system**: The points have an associated coordinate system. A typical example of a coordinate system is
```
Comment (qued) [inline docs/source/metadata.rst:75]: Same with system:
```suggestion
* ``system``: The points have an associated coordinate system. A typical example of a coordinate system is
```

File: docs/source/metadata.rst [hunk 13]
```diff
@@ -8,29 +12,67 @@ or an e-mail with a given subject line.
...
+Coordinates
+-----------
 
 Some document types support location data for the elements, usually in the form of bounding boxes.
 If it exists, an element's location data is available with ``element.metadata.coordinates``.
 
 The ``coordinates`` property of an ``ElementMetadata`` stores:
 
-* points: These specify the corners of the bounding box starting from the top left corner and
+* **points**: These specify the corners of the bounding box starting from the top left corner and
```
Comment (qued) [inline docs/source/metadata.rst:72]: Points should probably be monospace font, since it refers to a keyword in the code:
```suggestion
* ``points`` : These specify the corners of the bounding box starting from the top left corner and
```

File: docs/source/metadata.rst [hunk 14]
```diff
@@ -49,18 +91,56 @@ returned. If the ``in_place`` flag is ``False``, only the altered coordinates ar
...
 	print(element.metadata.coordinates.system.orientation)
 	print(element.metadata.coordinates.system.width)
 	print(element.metadata.coordinates.system.height)
 
+###########################################
+Additional Metadata Fields by Document Type
+###########################################
+
++-------------------------+---------------------+--------------------------------------------------------+
+| ``Field Name``          | Applicable Doc Types| Short Description                                      |
```
Comment (qued) [inline docs/source/metadata.rst:113]: Field Name should not be monospace, because it isn't referring to code:
```suggestion
| Field Name          | Applicable Doc Types| Short Description            |
```

File: docs/source/bricks/embedding.rst [hunk 15]
```diff
@@ -1,21 +1,21 @@
-########
+#########
 Embedding
-########
+#########
 
 EmbeddingEncoder classes in ``unstructured`` use document elements detected
```
Comment (qued) [inline docs/source/bricks/embedding.rst:5]: Here I would not refer to "EmbeddingEncoder" classes in camel case like that. It makes it sound like a Python class, but we don't define a class with that name. I would either go with something descriptive in plain English like "Embedding encoder classes..." or refer to the actual base class `BaseEmbeddingEncoder` (and in that case use monospace font to indicate this is actually something from the code).

**Review by qued — COMMENTED** [review]
Some nits mostly related to when to use monospace (code) font.

File: docs/source/metadata.rst [hunk 16]
```diff
@@ -8,29 +12,65 @@ or an e-mail with a given subject line.
...
+| coordinates                 | XY Bounding Box Coordinates                              | See notes below for further details about the bounding box.                                                                                                                                                                                                                                 |
++-----------------------------+----------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| parent_id                   | Element Hierarchy (Parent ID)                            | Hierarchies are determined by a combination of a ruleset and element category depth. The current ruleset sets a parent ID if a title element follows a header element or any other element follows a title element.                                                                         |
```
Comment (cragwolfe) [inline docs/source/metadata.rst:37]:
```suggestion
| parent_id                   | Element Hierarchy (Parent ID)                            | `parent_id` may be used to infer where an element resides within the overall hierarchy of a document. For instance, a `NarrativeText` element may have a `Title` element as a parent (a "sub-title"), which in turn may have another `Title` element as its parent (a "title).                                                                          |
```

File: docs/source/metadata.rst [hunk 17]
```diff
@@ -8,29 +12,65 @@ or an e-mail with a given subject line.
...
+|                             |                                                          | The ID is also set if the element follows an element of the same category and the category_depth is greater than the category depth of the element it follows. Hierarchies enable more robust chunking configurations.                                                                      |
```
Comment (cragwolfe) [inline docs/source/metadata.rst:38]:
```suggestion
```

File: docs/source/metadata.rst [hunk 18]
```diff
@@ -8,29 +12,65 @@ or an e-mail with a given subject line.
...
++-----------------------------+----------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| category_depth              | Element Depth relative to                                | Category depth is the depth of an element relative to other elements of the same category. It's set by a document partitioner and enables the hierarchy post-processor to compute more accurate hierarchies.                                                                                |
+|                             | other elements of the same category                      | Category depth is set using native document hierarchies (e.g., h1, h2, h3 or the indentation level of a bulleted list in a word document).                                                                                                                                                  |
```
Comment (cragwolfe) [inline docs/source/metadata.rst:41]:
```suggestion
|                             | other elements of the same category                      | Category depth may be set using native document hierarchies, e.g.  reflecting \<H1>, \<H2>, or `\<H3>` tags within an HTML document or the indentation level of a bulleted list item in a Word document.                                                                                                                                         |
```

File: docs/source/metadata.rst [hunk 19]
```diff
@@ -89,13 +168,86 @@ Webpages
 Elements from webpages will include a ``url`` metadata field, corresponding to the URL for the webpage.
 
 
+##############################
+Data Connector Metadata Fields
+##############################
+
```
Comment (cragwolfe) [inline docs/source/metadata.rst:174]:
```suggestion

Documents processed through unstructured-ingest connectors include additional document metadata. These additional fields only ever appear if the source document was processed by a connector.
```

**Review by cragwolfe — APPROVED** [review]
⚡
